pkg/storages/gcs/folder.go

```diff
@@ -282,12 +282,16 @@ func (folder *Folder) ReadObject(objectRelativePath string) (io.ReadCloser, erro
 }
 
 func (folder *Folder) PutObject(name string, content io.Reader) error {
-	tracelog.DebugLogger.Printf("Put %v into %v\n", name, folder.path)
-	object := folder.BuildObjectHandle(folder.joinPath(folder.path, name))
-
 	ctx, cancel := folder.createTimeoutContext()
 	defer cancel()
 
+	return folder.PutObjectWithContext(ctx, name, content)
+}
+
+func (folder *Folder) PutObjectWithContext(ctx context.Context, name string, content io.Reader) error {
+	tracelog.DebugLogger.Printf("Put %v into %v\n", name, folder.path)
```

Comment:
Actually, I found a regression here: now we won't use the timeout context as we used to do. I suggest passing the ctx to the `folder.createTimeoutContext()`.